### PR TITLE
feat:add options for password generation (Issue #318)

### PR DESCRIPTION
**Pull request by qmohitsingh**

### CODE REVIEW

File: include/faker-cxx/Internet.h [hunk 1]
```diff
@@ -30,6 +30,18 @@ enum class IPv4Class
     C
 };
 
+// Define the PasswordOptions struct
+struct PasswordOptions {
+    bool upperLetters;
+    bool lowerLetters;
+    bool numbers;
+    bool symbols;d
```
Comment (cieslarmichal) [inline include/faker-cxx/Internet.h:38]: typo ";d" ?

File: include/faker-cxx/Internet.h [hunk 2]
```diff
@@ -30,6 +30,18 @@ enum class IPv4Class
...
+
+    PasswordOptions(bool upper = true, bool lower = true, bool num = true, bool sym = true)
```
Comment (cieslarmichal) [inline include/faker-cxx/Internet.h:40]: please remove constructor and add default values to struct's fileds like bool upperLetters = true;

File: include/faker-cxx/Internet.h [hunk 3]
```diff
@@ -30,6 +30,18 @@ enum class IPv4Class
     C
 };
 
+// Define the PasswordOptions struct
```
Comment (cieslarmichal) [inline include/faker-cxx/Internet.h:33]: remove not needed comments

File: src/modules/internet/Internet.cpp [hunk 4]
```diff
@@ -93,21 +93,47 @@ std::string Internet::exampleEmail(std::optional<std::string> firstName, std::op
...
-std::string Internet::password(int length)
+std::string Internet::password(int length, PasswordOptions options)
 {
-    const std::string passwordCharacters =
-        "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789~`!@#$%^&*()_-+={[}]|:;\"'<,>.?/";
+    std::string characters;
 
-    std::string password;
+    if (options.upperLetters) {
+        characters += "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
```
Comment (cieslarmichal) [inline src/modules/internet/Internet.cpp:101]: those upperLetters etc are already defined in src/modules/string/data/Characters.h, you can use them and define not already defined group there

File: src/modules/internet/Internet.cpp [hunk 5]
```diff
@@ -93,21 +93,47 @@ std::string Internet::exampleEmail(std::optional<std::string> firstName, std::op
...
+//    std::string password;
+//
+//    for (int i = 0; i < length; i++)
+//    {
+//        password += Helper::arrayElement<char>(passwordCharacters);
+//    }
+//
+//    return password;
+//}
+
```
Comment (cieslarmichal) [inline src/modules/internet/Internet.cpp:136]: remove commented code

**Review by cieslarmichal — CHANGES_REQUESTED** [review]

File: include/faker-cxx/Internet.h (hunk 1 above)
Comment (qmohitsingh) [reply include/faker-cxx/Internet.h]: yes! removed

File: include/faker-cxx/Internet.h (hunk 2 above)
Comment (qmohitsingh) [reply include/faker-cxx/Internet.h]: fixed

File: include/faker-cxx/Internet.h (hunk 3 above)
Comment (qmohitsingh) [reply include/faker-cxx/Internet.h]: removed

File: src/modules/internet/Internet.cpp (hunk 4 above)
Comment (qmohitsingh) [reply src/modules/internet/Internet.cpp]: done

File: src/modules/internet/Internet.cpp (hunk 5 above)
Comment (qmohitsingh) [reply src/modules/internet/Internet.cpp]: removed

**Review by cieslarmichal — APPROVED** [review]

**Pull request merged**